onnxruntime/core/graph/graph_utils.cc

```diff
@@ -1,21 +1,85 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
 
 #include "core/graph/graph_utils.h"
+#include "core/graph/graph.h"
+#include "core/framework/tensorprotoutils.h"
 
 namespace onnxruntime {
 
-namespace utils {
-  // fusion is only done for ONNX domain ops
-  bool IsSupportedOptypeVersionAndDomain(const Node& node,
-                                         const std::string& op_type,
-                                         ONNX_NAMESPACE::OperatorSetVersion version,
-                                         const std::string& domain) {
-    if (node.OpType() != op_type ||
-        node.Op()->Deprecated() || node.Op()->SinceVersion() != version ||
-        (!node.Domain().empty() && node.Domain() != domain)) {
+namespace graph_edit_utils {
+// fusion is only done for ONNX domain ops
+bool IsSupportedOptypeVersionAndDomain(const Node& node,
+                                       const std::string& op_type,
+                                       ONNX_NAMESPACE::OperatorSetVersion version,
+                                       const std::string& domain) {
+  if (node.OpType() != op_type ||
+      node.Op()->Deprecated() || node.Op()->SinceVersion() != version ||
+      (!node.Domain().empty() && node.Domain() != domain)) {
+    return false;
+  }
+  return true;
+}
+
+bool IsConstantInputsNode(const Graph& graph, const Node& node) {
+  if (node.GetInputEdgesCount() > 0) {
+    return false;
+  }
+  const ONNX_NAMESPACE::TensorProto* initializer = nullptr;
+  for (const auto* input_def : node.InputDefs()) {
+    if (!graph.GetInitializedTensor(input_def->Name(), initializer)) {
       return false;
     }
-    return true;
   }
+  return true;
 }
 
-}  // namespace onnxruntime
\ No newline at end of file
+Status BuildSubgraph(const Graph& graph,
+                     const std::vector<NodeIndex>& subgraph_nodes,
+                     Graph& subgraph) {
+  // Add nodes and initializers to subgraph.
+  // TODO Can we directly copy the node instead of re-creating it?
+  for (auto& node_index : subgraph_nodes) {
```

Comment:
Probably not as the NodeIndex will differ, but perhaps there could be a Graph::AddNode(const Node&) method so this node copy logic is more likely to be re-used.

Reply:
Sounds good, will add a static method in the utils.

Reply:
The buildSubgraph method is no longer used in the latest code, but I think it is useful code for other scenarios. Shall I leave it here or just remove it?

Reply:
Will do.